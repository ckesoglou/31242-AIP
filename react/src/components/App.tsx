import React from "react";
import "../assets/css/App.css";
import {
  Switch,
  Route,
  BrowserRouter as Router,
  Redirect,
} from "react-router-dom";
import Login from "../pages/login";
import SignUp from "../pages/signup";
import UserProfile from "../pages/userprofile";
import Home from "../pages/home";
import { ProtectedRoute } from "./protected-route";
import { UserContext } from "./user-context";

type AppState = {
  user: Object;
  updateUser: (newUser: Object) => void;
};

class App extends React.Component {
  state: AppState = {
    user: { name: "?" },
    updateUser: (newUser: Object) => {
      this.setState({ user: newUser });
    },
  };

  render() {
    return (
      <Router>
        <UserContext.Provider value={this.state}>
          <Switch>
            <ProtectedRoute
              path="/user"
              component={UserProfile}
            ></ProtectedRoute>
            <Route path="/signup" component={SignUp} />
            <Route path="/login" component={Login} />
            <Route path="/home" component={Home} />
            <Route exact path="/">
              <Redirect to="/home" />
            </Route>
          </Switch>
        </UserContext.Provider>
      </Router>
    );
  }
}

export default App;
