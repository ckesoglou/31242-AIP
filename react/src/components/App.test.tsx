import React from "react";
import { shallow, mount } from "enzyme";
import App from "./App";

describe("<App />", () => {
  it("should render correctly", () => {
    const wrapper = shallow(<App />);
    
    expect(wrapper).toMatchSnapshot();
  });
});
