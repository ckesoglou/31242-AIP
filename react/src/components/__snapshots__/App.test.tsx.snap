// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render correctly 1`] = `
<BrowserRouter>
  <ContextProvider
    value={
      Object {
        "updateUser": [Function],
        "user": Object {
          "name": "?",
        },
      }
    }
  >
    <Switch>
      <ProtectedRoute
        component={[Function]}
        path="/user"
      />
      <Route
        component={[Function]}
        path="/signup"
      />
      <Route
        component={[Function]}
        path="/login"
      />
      <Route
        component={[Function]}
        path="/home"
      />
      <Route
        exact={true}
        path="/"
      >
        <Redirect
          to="/home"
        />
      </Route>
    </Switch>
  </ContextProvider>
</BrowserRouter>
`;
